src/main/java/by/exadel/internship/controller/AuthController.java

```diff
@@ -0,0 +1,58 @@
+package by.exadel.internship.controller;
+
+import by.exadel.internship.config.jwt.JwtService;
+import by.exadel.internship.entity.UserDetailsImpl;
+import by.exadel.internship.pojo.JwtResponse;
+import by.exadel.internship.pojo.LoginRequest;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+
+@Api(tags = "Authorization controller")
+public class AuthController {
+    private final AuthenticationManager authenticationManager;
+
+    private final JwtService jwtService;
+
+    @PostMapping("/signIn")
+    @ApiOperation("Authorize method")
+    public ResponseEntity<?> authUser(@RequestBody LoginRequest loginRequest){
+        splitEmail(loginRequest);
+        Authentication authentication = authenticationManager
+                .authenticate(new UsernamePasswordAuthenticationToken(
+                        loginRequest.getLogin(),
```

Comment:
Now you give token for any user without checking login in DB and checking password. We can't provide access to application to user who is not related to our system.

Reply:
As i understand, user should send his login and password, and if we find that user in DB we generate token and send its token to user. And other user can't get this token

Reply:
It's pretty simple to check your assumptions. Did you test with not existing login or invalid password?

Reply:
It should throw exception on invalid credentials while executing authenticationManager.authenticate if everything is configured correct

Reply:
We have Bad credentials exception, but browser error looks like 400 Bad Request. Should this be changed?

Reply:
>  Should this be changed?
It's tricky question. 
From my point of view error should be 401 since it's related to problem with user authentication. 400 error is not about access to resource. I consider it doesn't matter how user try to get access by token, by Basic auth or by credentials from form all of them are about authentication.
I guess spring can throw 400 because you've already inside Controller.  
But I'd like to listen what @SergeyKoval @viktarkolbik think about it. 400 or 401?

Reply:
400 should NOT be returned in case of invalid authentication. In any scenario

Reply:
I added BadCreditalException Handler and now when we throw exception we get a 401 status. Fixed